.github/workflows/formatting.yml

```diff
@@ -0,0 +1,103 @@
+name: formatting-check
+
+on:
+  push:
+    # Only run on pushes to pull request branches
+    branches-ignore:
+      - 'master'
+      - 'master-*'
+  pull_request:
+
+jobs:
+
+  # Checking for correct formatting of branch for C code changes
+  check-formatting:
+    name: Formatting Check (clang 9)
```

Comment:
ubuntu 18.04 also offers clang-10 now. Not sure if that would offer any benefits for this. Just a FYI

Reply:
Like this test? https://github.com/roligugus/suricata/actions/runs/169573386 ;)

Reply:
There are only a few clang 10 settings. See footnotes in https://github.com/roligugus/suricata/blob/clang-format-proposal-v4/doc/devguide/codebase/code-style.rst

Two settings that could be interesting:
- IndentGotoLabels, see https://github.com/roligugus/suricata/blob/clang-format-proposal-v4/doc/devguide/codebase/code-style.rst#id16
  In case you prefer the nested goto labels leftmost bound
- AllowShortBlocksOnASingleLine, see https://github.com/roligugus/suricata/blob/clang-format-proposal-v4/doc/devguide/codebase/code-style.rst#id6
  Could allow for finer-grained setting.

Reply:
@victorjulien Unless you care about these two settings, I'd recommend to "keep it as old as acceptable/possible". I.e. stay with the slightly older clang-9 settings just to support older dists that might not have clang-10 conveniently packaged up. Neither debian or ubuntu are and issue. fedora would require v32 as v31 only has convenient clang-9 package.

clang-9 does not have these settings yet, but the behaviour is:
- IndentGotoLabels: true (it indents them)
- AllowShortBlocksOnASingleLine: Never (see below)

IndentGotoLabels has an example in code-style.rst, but here is the description from the official clang-format page for your convenience for this comment:

**IndentGotoLabels (bool)**
Indent goto labels.

When false, goto labels are flushed left.
```
true:                                  false:
int f() {                      vs.     int f() {
  if (foo()) {                           if (foo()) {
  label1:                              label1:
    bar();                                 bar();
  }                                      }
label2:                                label2:
  return 1;                              return 1;
}                                      }
```

FYI, AllowShortBlocksOnASingleLine is not described in code-style.rst, so here is the description from the official clang-format page (do I wish I would not have to copy it and could link to the paragraph instead):

**AllowShortBlocksOnASingleLine (ShortBlockStyle)**
Dependent on the value, while (true) { continue; } can be put on a single line.

Possible values:

SBS_Never (in configuration: Never) Never merge blocks into a single line.
```
while (true) {
}
while (true) {
  continue;
}
```

SBS_Empty (in configuration: Empty) Only merge empty blocks.
```
while (true) {}
while (true) {
  continue;
}
```
SBS_Always (in configuration: Always) Always merge short blocks into a single line.
```
while (true) {}
while (true) { continue; }
```

Reply:
Sticking to clang 9 for now as mentioned elsewhere.